texar/torch/modules/decoders/gpt2_decoder.py

```diff
@@ -256,3 +258,43 @@ def default_hparams():
             'name': 'gpt2_decoder',
             '@no_typecheck': ['pretrained_model_name'],
         }
+
+    def forward(self,  # type: ignore
+                inputs: Optional[torch.Tensor] = None,
+                sequence_length: Optional[torch.LongTensor] = None,
+                memory: Optional[torch.Tensor] = None,
+                memory_sequence_length: Optional[torch.LongTensor] = None,
+                memory_attention_bias: Optional[torch.Tensor] = None,
+                context: Optional[torch.Tensor] = None,
+                context_sequence_length: Optional[torch.LongTensor] = None,
+                helper: Optional[Helper] = None,
+                decoding_strategy: str = 'train_greedy',
+                max_decoding_length: Optional[int] = None,
+                impute_finished: bool = False,
+                infer_mode: Optional[bool] = None,
+                beam_width: Optional[int] = None,
+                length_penalty: float = 0.,
+                **kwargs) \
+            -> Union[
+                TransformerDecoderOutput,
+                Tuple[TransformerDecoderOutput, torch.LongTensor],
+                Dict[str, torch.Tensor]]:
+        r"""Performs decoding. Has exact the same functionality with
+        :meth:`texar.torch.modules.TransformerDecoder.forward`. Please refer to
+        it for the detailed information.
```

Comment:
Editted a little bit:

Performs decoding. Has exact the same interfaces with :meth:`texar.torch.modules.TransformerDecoder.forward`. Please refer to it for the detailed usage.